### DISCUSSION TITLE
filter to exclude current id

### DISCUSSION THREAD

**Discussion by kamil-malinski** [directus/directus#12931] [Archive]
Hi!

I have a collection of articles. An article can be either of type „article“ or of type „redirect“. If the article is of type „redirect“ the user needs to select another article he wants the article to redirect to. Therefore I have a n:1 field where the user can select another article. Is it possible to filter for articles that are not the current article so the user won’t create an infinite redirect loop? So something like

„redirect_to_id not equal $current_id“

Cheers

Kamil

**Comment by rijkvanzanten** [comment]
There's currently no support for dynamic variables in the configured filter based on the value of the current item scope 👍🏻

**Reply by nervusdm** [reply]
Oh, I was looking for this. Filter should use dynamic variable. It's not even on the road map sadly.

**Reply by rijkvanzanten** [reply]
> It's not even on the road map sadly.

The road map on the website is just a bigger overview of things we'd like to tackle. It's not an all-inclusive list of every little feature that's going to be worked on. For smaller improvements like this, we base it on popularity in the [feature requests](https://github.com/directus/directus/discussions/categories/ideas?discussions_q=sort%3Atop+category%3AIdeas). Looking there, you'd find that this is actually a very often requested feature: https://github.com/directus/directus/discussions/2993 and therefore something that will most likely come pretty soon

**Reply by rijkvanzanten** [reply]
oh and to add to that, it _is_ on the road map. This is one of the enhancements done to the permissions/validation engine

<img width="306" alt="CleanShot 2022-08-09 at 11 00 35@2x" src="https://user-images.githubusercontent.com/9141017/183685451-88e00811-b5fa-479c-963e-5837680ca70b.png">
